cachito/web/models.py

```diff
@@ -503,13 +565,24 @@ def _state_to_json(state):
             rv["state_history"] = states
 
             package_to_deps = {}
+            # This dictionary will have the keys as dependency names and the values as sets of
+            # dependency IDs. This is used to quickly see if a dependency has been added to the
+            # root "dependencies" array in the returned JSON. This uses a dictionary instead of
+            # a list to avoid an O(n) operation for every dependency.
+            dep_name_to_dep_ids = {}
```

Comment:
If dependency IDs are unique, won't a simple set work?
```python
already_seen_dependencies = set()
...
if req_dep.dependency.id not in already_seen_dependencies:
  ...
  already_seen_dependencies.add(req_dep.dependency.id)
```

Reply:
You're right. The time complexity of an `in` operation on a set averages to O(n) like accessing a dictionary key. That makes sense. I'll do that.